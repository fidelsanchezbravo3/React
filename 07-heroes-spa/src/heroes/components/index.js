
export * from './HeroeCard';
export * from './HeroeList';