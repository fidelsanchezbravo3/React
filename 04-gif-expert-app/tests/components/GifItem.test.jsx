import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => { 

    const titulo = 'One Punch';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Debe hacer match con el snapshot', () => { 

        const { container } = render( <GifItem title={ titulo } url={ url }/> );
        expect( container ).toMatchSnapshot();

     });

     test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render( <GifItem title={ titulo } url={ url } /> );

        // expect( screen.getByRole('img').src ).toBe( url );
        // expect( screen.getByRole('img').alt ).toBe( titulo );
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( titulo );

     });

     test('Debe de mostrar el titulo en el componente', () => { 

        render( <GifItem title={ titulo } url={ url } /> );
        

      });
 });