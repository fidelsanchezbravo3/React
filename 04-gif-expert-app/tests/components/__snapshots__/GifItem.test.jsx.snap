// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
